cve_bin_tool/extractor.py

```diff
@@ -74,32 +74,43 @@ def can_extract(self, filename):
                 return True
         return False
 
-    @staticmethod
-    async def extract_file_tar(filename, extraction_path):
+    def tar_member_filter(self, members, extraction_path):
+        """Generator function to serve as a backported filter for tarfile extraction
+        based on https://docs.python.org/3/library/tarfile.html#examples
+        """
+        for tarmember in members:
+            if tarmember.isfile() and str(
+                Path(extraction_path, tarmember.name).resolve()
+            ).startsWith(extraction_path):
+                yield tarmember
+
+    async def extract_file_tar(self, filename, extraction_path):
         """Extract tar files"""
+
+        # make sure we have full path for later checks
+        extraction_path = str(Path(extraction_path).resolve())
         with ErrorHandler(mode=ErrorMode.Ignore) as e:
+            tar = tarfile.open(filename)
             # Python 3.12 has a data filter we can use in extract
             # tarfile has this available in older versions as well
             if hasattr(tarfile, "data_filter"):
-                await aio_unpack_archive(filename, extraction_path, filter="data")
+                tar.extractall(path=extraction_path, filter="data")  # nosec
+                # nosec line because bandit doesn't understand filters yet
+
+            # FIXME: the backported fix is not working on windows.
+            # this leaves the current (unsafe) behaviour so we can fix at least one OS for now
+            if sys.platform == "win32":
+                tar.extractall(path=extraction_path)  # nosec
+                tar.close()
```

Comment:
There's a call to tar.close after the next 'else' section that will be called. With this one here, targ.close gets called twice.

Reply:
thank you.  Absolutely non-sarcastically, this is why I lovecode review.  I'd originally had the close in multiple spots and though "oh I should move it below" and then I apparently didn't move them all. 🤦  Totally the sort of thing I miss when reading my own code.